src/components/price/service.ts

```diff
@@ -74,27 +74,57 @@ export class PriceService {
 
     isInputPriceValid(swap) {
         try {
-            const pairPrice = this.getPairPrice(swap.network, swap.outputNetwork);
+            const pairPrice = this.getPairPriceWithSpreadAndFee(swap.network, swap.outputNetwork);
 
             const inputDecimals = this.blockchainConfig[swap.network].decimals;
             const outputDecimals = this.blockchainConfig[swap.outputNetwork].decimals;
 
-            const priceInBig = mulDecimals(pairPrice, outputDecimals);
+            //locked Amount
+            const lockedAmountSlashed = divDecimals(swap.inputAmount, inputDecimals);
 
-            const outputAmount = mul(priceInBig, divDecimals(swap.inputAmount, inputDecimals));
+            //Desired Amount
+            const desiredAmountSlashed = divDecimals(swap.outputAmount, outputDecimals);
 
-            const pairSlippage = this.userConfig.PAIRS[`${swap.network}-${swap.outputNetwork}`].SLIPPAGE || 0;
+            //Current price 
+            const marketActualPrice = mul(lockedAmountSlashed, pairPrice);
 
-            const requestedAmount = mul(outputAmount, add(1, pairSlippage));
+            //MAX Allowed Slippage
+            const maxAllowedSlippage = mul(marketActualPrice, AppConfig.SLIPPAGE);
 
-            const outputAmountWithoutFee = sub(swap.outputAmount, mul(swap.outputAmount, AppConfig.FEE));
+            greaterThan(maxAllowedSlippage, sub(desiredAmountSlashed, marketActualPrice));
 
-            return greaterThan(outputAmountWithoutFee, requestedAmount);
         } catch (err) {
             return false;
         }
     }
 
+    adjustPrice(swap){
```

Comment:
Indentation is wrong. Also the funciton name is misleading because it's modifying the `inputAmount` and `adjustPrice` is not relevant.

Reply:
So the function can be removed from the price service and moved to the swap handler (since it's only being used there). A better name would be: `getCurerntInputAmount` or `getLatestInputAmount`